app/models/consultation_response.rb

```diff
@@ -102,6 +121,26 @@ def validate_answers
     end
   end
 
+  def notify_admin_if_profane
+    if self.is_approved == "under_review"
+      NotifyProfaneResponseEmailToAdminJob.perform_later(self)
+    end
+  end
+
+  def approve
+    self.approved_by_id = Current.user.id
+    self.is_approved = :acceptable
```

Comment:
`is_approved` can be a boolean field rather than having :acceptable, the naming convention is more readalbe.

Reply:
The workflow that we designed for handling profane responses required us to look at 3 different statuses and hence boolean was not opted as a choice. Instead this is declared as an Integer with values 0,1 and 2 for:

a) Acceptable.
b) Under Review.
c) Unacceptable.

Reply:
@kanth5 - understood your workflow. My problem is that the world `is_` indicates that the field is a boolean. 

I have two suggestions on the way forward. 

**Create a status column**
You could create a column called status, that indicates what is the current status of the response. The default can be `under_review` and later based on the admin actions, they can be moved to `acceptable` or `unacceptable`

**Use the is_approved column**
You can still use the `is_approved` column as a boolean, where-in if there the response is under_review, it can be left as `nil`

Reply:
Changed the 'is_approved' column name to 'response_status' at all the places in the code related to handling consultation responses both on the backend as well as front end.